File: plugins/document/scripts/pdfCreatorCmd.py
import win32print
import os.path
import time
import sys

if len(sys.argv) < 4:
    print 'wrong usage of this script. usage: %s {inputFile} {outFile} [--readOnly]' % os.path.dirname(__file__);
    sys.exit(1)

# build the command line
if sys.argv[-1] == '--readonly':
    readOnly = True
    inputFile = sys.argv[-3]
    outputFile = sys.argv[-2]
    commandParams = sys.argv[1:-3]
else:
    readOnly = False
    inputFile = sys.argv[-2]
    outputFile = sys.argv[-1]
    commandParams = sys.argv[1:-2]

commandParams.append('/NoStart')

inputFileExt = os.path.splitext(inputFile)[1].lower()
if readOnly and inputFileExt == '.pdf':
    commandParams.append('/IF"%s"' % inputFile)
    commandParams.append('/OF"%s.pdf"' % outputFile)
else:
    commandParams.append('/PF"%s"' % inputFile)

command = ' '.join(commandParams)

# make sure the default printer is set appropriately
PDF_CREATOR_PRINTER = 'PDFCreator'
if win32print.GetDefaultPrinter() != PDF_CREATOR_PRINTER:
    print 'setting default printer to %s' % PDF_CREATOR_PRINTER
    win32print.SetDefaultPrinter(PDF_CREATOR_PRINTER)

# execute the command
print '\ncommand: %s' % command
os.system(command)

# wait until the printer queue becomes empty
printer = win32print.OpenPrinter('PDFCreator')

while True:
    if len(win32print.EnumJobs(printer, 0, 1, 2)) == 0:
        break
    time.sleep(.5)

File: tags/svn+ssh:/kelev.kaltura.com/usr/local/kalsource/backend/server/tags/falcon_2012_08_20/plugins/document/scripts/pdfCreatorCmd.php
<?php
if ($argc < 4) {
	echo 'wrong usage of this script. usage: ' . $_SERVER ['SCRIPT_NAME'] . ' {inputFile} {outFile} [--readOnly]' . PHP_EOL;
	die ();
}
if (end ( $argv ) == '--readonly') {
	$readonly = true;
	$inputFile = $argv [$argc - 3];
	$outputFile = $argv [$argc - 2];
	$commandArgc = $argc - 4;
} else {
	$readonly = false;
	$inputFile = $argv [$argc - 2];
	$outputFile = $argv [$argc - 1];
	$commandArgc = $argc - 3;
}
$command = "";
for($i = 1; $i <= $commandArgc; $i ++) {
	$command .= $argv [$i].' ';
}

$command .= ' /NoStart ';
$inputFileExtension = pathinfo ( $inputFile, PATHINFO_EXTENSION );
$inputFileExtension = strtolower ( $inputFileExtension );

if (($readonly) && ($inputFileExtension == 'pdf')) {
	$command .= '/IF"' . $inputFile . '" /OF"' . $outputFile . '.pdf"';
} else {
	$command .= '/PF"' . $inputFile . '"';
}

echo "\ncommand: $command";
exec ( $command ); //TODO - return a value. and exit with this value
